src/model/channels.js

```diff
@@ -215,6 +223,62 @@ export {RouteDef}
  */
 const ChannelSchema = new Schema(ChannelDef)
 
+// "pre" is a middleware that will run before the update takes place, 'this' will contain the new channel details
+ChannelSchema.pre('save', async function (next) {
+  const timeout = this.timeout ?? +config.router.timeout
+  const kafkaRoutes = this.routes.filter(e => e.type === 'kafka')
+
+  let originalKafkaDetails = await ChannelModel.aggregate([
+    {$match: {_id: this._id}},
+    {
+      $project: {
+        timeout,
+        routes: {
+          $filter: {
+            input: '$routes',
+            as: 'route',
+            cond: {$eq: ['$$route.type', 'kafka']}
+          }
+        }
+      }
+    }
+  ])
+
+  // We need to cross reference the original, not-yet modified, routes
+  // against the incoming dirty routes to see if any were removed and if so remove them from the manager
+  if (Array.isArray(originalKafkaDetails)) originalKafkaDetails = originalKafkaDetails[0]
+  if (originalKafkaDetails && originalKafkaDetails.routes.length > 0) {
```

Comment:
is this an object or array ? could we be strict in typing ?

Reply:
done :)